sources/packages/backend/apps/api/src/route-controllers/application-exception/_tests_/e2e/application-exception.aest.approveException.e2e-spec.ts

```diff
@@ -0,0 +1,221 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import { DataSource } from "typeorm";
+import {
+  AESTGroups,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAESTToken,
+} from "../../../../testHelpers";
+import { ApplicationExceptionStatus } from "@sims/sims-db";
+import { ClientTypeBaseRoute } from "../../../../types";
+import * as faker from "faker";
+import { createFakeApplicationWithApplicationException } from "../application-exception-helper";
+import { TestingModule } from "@nestjs/testing";
+import { ZBClient } from "zeebe-node";
+
+describe(`${ClientTypeBaseRoute.AEST}-ApplicationExceptionAESTController(e2e)-approveException`, () => {
+  let app: INestApplication;
+  let appDataSource: DataSource;
+  let appModule: TestingModule;
+  let zbClient: ZBClient;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource, module } =
+      await createTestingAppModule();
+    app = nestApplication;
+    appDataSource = dataSource;
+    appModule = module;
+    zbClient = app.get(ZBClient);
+  });
+
+  it("Should be able to approve application exception when available.", async () => {
+    // Arrange
+    const application = await createFakeApplicationWithApplicationException(
+      ApplicationExceptionStatus.Pending,
+      appDataSource,
+      appModule,
+    );
+    const updateApplicationException = {
+      exceptionStatus: ApplicationExceptionStatus.Approved,
+      noteDescription: faker.lorem.text(10),
+    };
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .patch(
+        `/aest/application-exception/${application.applicationException.id}`,
+      )
+      .send(updateApplicationException)
+      .auth(
+        await getAESTToken(AESTGroups.BusinessAdministrators),
+        BEARER_AUTH_TYPE,
+      )
+      .expect(HttpStatus.OK);
+
+    expect(zbClient.publishMessage).toHaveBeenCalledWith(
+      expect.objectContaining({
```

Comment:
Can you please check if it cannot be simplified as 
```ts
expect(zbClient.publishMessage).toHaveBeenCalledWith({
    correlationKey: application.applicationException.id.toString(),
    name: "application-exception-verified",
    variables: {
      applicationExceptionStatus: "Approved",
});
```

Reply:
There's some properties I'm not checking because I think they not important for the test:
![image](https://user-images.githubusercontent.com/78114138/227663911-07ff5699-b7ca-4ceb-86e7-55e6082b6e91.png)

Reply:
To be honest I would be ok having the complete object since `value: 10` is also part of the payload, but I am ok also with this way since the key properties are validated and enough to ensure that the specific call happened.